### PR TITLE
feat: blend point cloud classification colors with RGB

### PR DESCRIPTION
**Pull request by larsmoa**
# Description

Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context. 
List any dependencies that are required for this change.

# Checklist:

Here is a checklist that should completed before merging this given feature. 
Any shortcomings from the items below should be explained and detailed within the contents of this PR.

- [x] I am proud of this feature.
- [x] I have performed a self-review of my own code.
- [x] I have manually tested this for different scenarios (different models, formats, devices, browsers).
- [x] I have commented my code in hard-to-understand areas.
- [x] I have made corresponding changes to the public documentation.
- [x] I have added unit and visuals tests to prove that my feature works to the best of my ability.
- [x] I have refactored the code for readability to the best of my ability.
- [x] I have checked that my changes do not introduce regressions in the public documentation.
- [x] I have outlined any known defects / lacking capabilities in the contents of this PR.
- [x] I have listed any remaining work that should follow this PR in the description or jira/miro/etc.
- [x] I have added TSDoc to any public facing changes.
- [x] I have added "developer documentation" in any package README.md that is related / applicable to this PR.
- [x] I have noted down and am currently tracking any technical debt introduced in this PR.
- [x] I have thoroughly thought about the architecture of this implementation.
- [x] I have asked peers to test this feature.

### CODE REVIEW

File: viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert [hunk 1]
```diff
@@ -548,7 +548,11 @@ void main() {
 	#elif defined color_type_point_index
 		vColor = indices.rgb;
 	#elif defined color_type_classification
-		vColor = classification.rgb;
+		vec3 rgb = getRGB();
+		vColor = rgb == vec3(0.0, 0.0, 0.0) 
+			? classification.rgb 
+			: mix(classification.rgb, rgb, 0.6);
+			// :
```
Comment (haakonflatval-cognite) [inline viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert:555]:
```suggestion
```

**Review by haakonflatval-cognite — COMMENTED** [review]
LGTM, but what is that thing

**Review by haakonflatval-cognite — APPROVED** [review]
LGTM :+1:

File: viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert [hunk 2]
```diff
@@ -548,7 +548,10 @@ void main() {
 	#elif defined color_type_point_index
 		vColor = indices.rgb;
 	#elif defined color_type_classification
-		vColor = classification.rgb;
+		vec3 rgb = getRGB();
+		vColor = rgb == vec3(0.0, 0.0, 0.0) 
+			? classification.rgb 
+			: mix(classification.rgb, rgb, 0.6);
```
Comment (chingyulin) [inline viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert:554]: I guess 0.6 is mix ratio, should it be more explicit? Or maybe hold it as a variable

**Review by chingyulin — APPROVED** [review]
LGTM

File: viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert (hunk 2 above)
Comment (larsmoa) [reply viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert]: I think ok for now - there's a lot of hardcoded values in there... But I changed the constant a bit to emphasise classification color.

File: viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert [hunk 3]
```diff
@@ -350,6 +350,12 @@ float getIntensity() {
 	return w;
 }
 
+vec3 getRgbWithIntensityFallback() {
```
Comment (haakonflatval-cognite) [inline viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert:353]: What is the reason we want intensity as fallback? It's not always available either, as far as I know, and I've seen some confusion earlier as to what range it lies in, but not sure if that's still a real issue.

Comment (larsmoa) [reply viewer/packages/pointclouds/src/potree-three-loader/rendering/shaders/pointcloud.vert:353]: Just a last resort. There are quite a few cases where RGB isn't available, and in almost all of these intensity will be available I think. Range might be an issue, but it improves situation slightly I guess.

**Pull request merged**